src/SAWScript/Crucible/LLVM/Builtins.hs

```diff
@@ -1975,8 +1986,13 @@ llvm_alloc_with_size sz lty =
     Crucible.Mutable
     (Just (Crucible.toBytes sz))
     Nothing
+    LLVMAllocSpecNoInitialization
     lty
 
+llvm_alloc_sym_init :: L.Type -> LLVMCrucibleSetupM (AllLLVM SetupValue)
```

Comment:
I'm curious: does this work in practice when you apply in an a specification that is assumed with `llvm_unsafe_assume_spec`? The reason I ask is that I'm doing something quite similar to this as part of ongoing work in #1461, and I couldn't make it work without making some nontrivial changes to the way `doInvalidate` works in `crucible`.

I suppose this is a long-winded way of asking for some more test cases using `llvm_alloc_sym_init`. I'm not sure if you plan to do this as part of this PR or as part of a later PR, however.